wazp/callbacks/roi.py

```diff
@@ -550,62 +642,143 @@ def save_rois_and_update_status_alert(
+        # Load saved ROIs from the metadata file, if any
+        try:
+            rois_in_file = utils.load_rois_from_yaml(metadata_path)
+        except FileNotFoundError:
+            alert_msg = f"Could not find {metadata_path.name}"
             alert_color = "danger"
-            return dash.no_update, alert_msg, alert_color, True
+            return alert_msg, alert_color, True
+        except KeyError:
+            rois_in_file = []
 
-        # Get the stored ROI shapes for this video
+        # Get the app's ROI shapes for this video
+        rois_in_app = []
         if video_name in roi_storage.keys():
-            roi_shapes = roi_storage[video_name]["shapes"]
-            rois_to_save = [
-                utils.stored_shape_to_yaml_entry(shape) for shape in roi_shapes
-            ]
-        else:
-            rois_to_save = []
+            rois_in_app = roi_storage[video_name]["shapes"]
 
-        if rois_to_save:
-            if trigger == "roi-storage.data" and rois_to_save == saved_rois:
-                # This means that the ROIs have just been loaded
-                alert_msg = f"Loaded ROIs from {metadata_filepath.name}"
-                alert_color = "success"
-                return dash.no_update, alert_msg, alert_color, True
-            elif trigger == "roi-storage.data" and rois_to_save != saved_rois:
-                # This means that the ROIs have been modified
-                # in respect to the metadata file
-                alert_msg = "Detected unsaved changes to ROIs."
-                alert_color = "warning"
-                return dash.no_update, alert_msg, alert_color, True
+        if not rois_in_app:
+            alert_color = "light"
+            if rois_in_file:
+                alert_msg = (
+                    f"Found {len(rois_in_file)} ROIs saved in "
+                    f"'{metadata_path.name}'. Click the "
+                    "'Load all from file' button to load them."
+                )
             else:
-                if save_clicks > 0:
-                    # This means that the user wants to save the ROIs
-                    # This overwrites any previously saved ROIs
-                    with open(metadata_filepath, "w") as yaml_file:
-                        metadata["ROIs"] = rois_to_save
-                        yaml.safe_dump(metadata, yaml_file)
-                    alert_msg = f"Saved ROIs to {metadata_filepath.name}"
-                    alert_color = "success"
-                    return (
-                        metadata_filepath.as_posix(),
-                        alert_msg,
-                        alert_color,
-                        True,
-                    )
-                else:
-                    return dash.no_update, dash.no_update, dash.no_update, True
+                alert_msg = (
+                    "No ROIs to save or to load. "
+                    "Draw some ROIs on the video frame."
+                )
```

Comment:
I think for this message I'd remove the 'Draw..' bit and just say 'No ROIs defined in the metadata file' (the 'save or load' was a bit confusing)